js/adapt-contrib-bookmarking.js

```diff
@@ -1,236 +1,234 @@
-define([
-  'core/js/adapt'
-], function(Adapt) {
-
-  var Bookmarking = Backbone.Controller.extend({
-
-    bookmarkLevel: null,
-    watchViewIds: null,
-    watchViews: [],
-    restoredLocationID: null,
-    currentLocationID: null,
-
-    initialize: function () {
-      this.listenToOnce(Adapt, 'router:location', this.onAdaptInitialize);
-    },
-
-    onAdaptInitialize: function() {
-      if (!this.checkIsEnabled()) return;
-      this.setupEventListeners();
-      this.checkRestoreLocation();
-    },
-
-    checkIsEnabled: function() {
-      var courseBookmarkModel = Adapt.course.get('_bookmarking');
-      if (!courseBookmarkModel || !courseBookmarkModel._isEnabled) return false;
-      if (!Adapt.offlineStorage) return false;
-      return true;
-    },
-
-    setupEventListeners: function() {
-      this._onScroll = _.debounce(this.checkLocation.bind(this), 1000);
-      this.listenTo(Adapt, {
-        'menuView:ready': this.setupMenu,
-        'pageView:preRender': this.setupPage
-      });
-    },
-
-    checkRestoreLocation: function() {
-      this.restoredLocationID = Adapt.offlineStorage.get('location');
-
-      if (!this.restoredLocationID || this.restoredLocationID === 'undefined') return;
-
-      this.listenToOnce(Adapt, 'pageView:ready menuView:ready', this.restoreLocation);
-    },
+import Adapt from 'core/js/adapt';
+
+class Bookmarking extends Backbone.Controller {
+
+  initialize() {
+    this.bookmarkLevel = null;
+    this.watchViewIds = null;
+    this.watchViews = [];
+    this.restoredLocationID = null;
+    this.currentLocationID = null;
+    this.listenToOnce(Adapt, 'router:location', this.onAdaptInitialize);
+  }
+
+  onAdaptInitialize() {
+    if (!this.checkIsEnabled()) return;
+    this.setupEventListeners();
+    this.checkRestoreLocation();
+  }
+
+  checkIsEnabled() {
+    const courseBookmarkModel = Adapt.course.get('_bookmarking');
+    if (!courseBookmarkModel || !courseBookmarkModel._isEnabled) return false;
+    if (!Adapt.offlineStorage) return false;
```

Comment:
this condition `!Adapt.offlineStorage` could go into the if statement on line 22

Reply:
Now that you mention it I'm actually struggling to think of any condition under which `Adapt.offlineStorage` wouldn't exist... might take that check out altogether.